src/coreclr/System.Private.CoreLib/src/System/RuntimeType.CoreCLR.cs

```diff
@@ -3363,18 +3370,47 @@ public override string? Namespace
 
         #region Attributes
 
-        public override Guid GUID
+        public override unsafe Guid GUID
         {
             get
             {
-                Guid result = default;
-                GetGUID(ref result);
+                TypeHandle th = GetNativeTypeHandle();
+                if (th.IsTypeDesc || th.AsMethodTable()->IsArray)
+                {
+                    return Guid.Empty;
+                }
+
+                Guid result;
+#if FEATURE_COMINTEROP
+                if (IsCOMObject)
```

Comment:
```suggestion
                if (IsGenericCOMObjectImpl())
```
I think it needs to be this condition.

Reply:
I think that is basically the same as checking if it is `__ComObject`. I'm still trying to work my way through this.

Reply:
Yes, this checks whether the RuntimeType points to `__ComObject` *MethodTable*. It is not the same as checking `this == typeof(__ComObject)`. The legacy Com interop is complicated.